scripts/version.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+#############################################################################
+# Copyright (c) 2019 Balabit
+# Copyright (c) 2018 Kokan <kokaipeter@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+BASEDIR=$(dirname "$0")
+
+cd $BASEDIR/../
+
+if [ -d .git ]; then
+  git describe --always --tags --dirty --abbrev=7 | cut -d'-' -f3- | tr '-' '+' | tr -d '\n'
```

Comment:
What do you think of `sed 's/^syslog-ng-//; s/-/+/g' | tr -d '\n'`?
I think removing `syslog-ng` explicitly is better than `cut -d'-' -f3-`.

Do we want `--tags`? I know, we sometimes forget to use annotated tags, but I think we should not.

We might also want to add `--match "syslog-ng-*"` in order to avoid non-version tags.

Reply:
As long as github does not do annotated tags I think `--tags` should be kept.
The current version supports if we may want to use different versioning system like (3.20.1).